File: h2.scala
import java.util._
object Main {
  val sc = new Scanner(System.in)
  val N = sc.nextInt

  def main(args: Array[String]): Unit = {
    def isOK(x: Double) = {
        x * (x * (x + 1) + 2) + 3 - N >= 0
    }

    var left = 0.0
    var right = 100.0
    var mid = (right + left) / 2

    while(right - left > 0.01){
        if(isOK(mid)){
            right = mid
        }
        else {
            left = mid
        }
        mid = (right + left) / 2
    }
    println(mid)
  }
}

File: h2.cs
using System;
class Program{
  static string[] input = Console.ReadLine().Split(' ');
  static double N = double.Parse(input[0]);

  static bool isLarger(double x){
    double result = x * (x * (x + 1) + 2) + 3;
    return result > N;
  }

  static void Main(string[] args){
    double left = 0;
    double right = 100;
    double mid = (left + right) / 2;
    
    while (right - left > 0.01){
      if(isLarger(mid)){
        right = mid;
      }
      else{
        left = mid;
      }
      mid = (left + right) / 2;   
    }

    Console.WriteLine(mid);
  }
}
